### PR TITLE
Changes for test server to work for Azure SDK happy path tests

### PR DESCRIPTION
**Pull request by brywang-msft**
## Overview
Bunch of fixes that enable happy path for Azure SDK tests. In other words, as long as the test doesn't expect an error, we should be compatible. The error cases need some additional work in PSSwagger.

## Major
- Server should be executable without parameters
- Serialize flattened properties in a "properties" sub-object (expected by AutoRest)
- Rework how we get metadata from specification
- When the output of a command is flattened, put the results back into the correct Page object
- Properly serialize command response

## Minor
- Add an event log logger for when named pipes fail
- Encoding fix
- Update xunit for VS2017 compat
- Fix a spinning issue in ConsoleServer.exe

### CODE REVIEW

File: PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Converters/LiveTestRequestConverter.cs [hunk 1]
```diff
@@ -181,5 +188,13 @@ private bool ConvertObject(object val, Type expectedType, JsonSerializer seriali
 
             return success;
         }
+
+        private void Log(string msg, params object[] args)
+        {
+            if (this.module.Logger != null)
+            {
+                this.module.Logger.Log(msg, args);
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Converters/LiveTestRequestConverter.cs]: >msg [](start = 39, length = 3)

msg can be null or whitespace, right? Consider adding a check on msg too.

File: PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        private IList<Logger> InternalLoggers { get; set; }
+
+        public CompositeLogger() : base(null, null)
+        {
+            this.InternalLoggers = new List<Logger>();
+        }
+
+        public CompositeLogger AddLogger(Logger logger)
+        {
+            this.InternalLoggers.Add(logger);
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs]: >logger [](start = 37, length = 6)

Assert on null value for logger.

File: PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            return this;
+        }
+
+        /// <summary>
+        /// Log a single message to output pipe.
+        /// </summary>
+        /// <param name="message">Message to log.</param>
+        public override void Log(string message, params object[] args)
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs]: >message [](start = 40, length = 7)

Consider adding checks on message param here and below functions too.

Comment (brywang-msft) [reply PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs]: e750658

File: PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs (hunk 2 above)
Comment (brywang-msft) [reply PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Logging/CompositeLogger.cs]: e750658

File: PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Converters/LiveTestRequestConverter.cs (hunk 1 above)
Comment (brywang-msft) [reply PSSwagger.LiveTestFramework/src/PSSwagger.LTF.Lib/Converters/LiveTestRequestConverter.cs]: e750658

**Pull request merged**